tests/jballerina-unit-test/src/test/resources/test-src/expressions/varref/error_binding_pattern_negative.bal

```diff
@@ -53,3 +53,46 @@ function testNestedNonRequiredFieldBinding() {
     error(m, message = n) = t[1];
     [i, error(m, message = n)] = t;
 }
+
+type SampleErrorData record {|
+    int code;
+    string reason;
+|};
+
+type SampleError error<SampleErrorData>;
+
+type TestRecord record {|
+    int errorNum2;
+    string errorString2;
+    SampleError err;
+|};
+
+type errorData record {|
+    int errNum;
+    string errReason;
+|};
+
+type MyError error<errorData>;
+
+function testWildcardBindingPatternWithErrorCause() {
+    SampleError sampleErr = error("Transaction Failure", error("Database Error"), code = 20,
+                            reason = "deadlock condition");
+
+    var error(message1, _, code = code1, reason = reason1) = sampleErr;
+
+    var error(_, _, code = code2, reason = reason2) = sampleErr;
+
+    SampleError error(message3, _, code = code3, reason = reason3) = sampleErr;
+
+    SampleError error(_, _, code = code4, reason = reason4) = sampleErr;
+
+    [int, [string, SampleError]] [errorNum1, [errorString1, error(message5, _, code = code5, reason = reason5)]] = [1234, ["ERROR", sampleErr]];
+
+    TestRecord testRecord = {errorNum2: 1223, errorString2: "ERROR", err: sampleErr};
+    TestRecord {errorNum2: _, errorString2: firstName, err: error(message6, _, code = code6, reason = reason6)} = testRecord;
+
+    MyError myErr = error("Illegal Return", sampleErr, errNum = 20, errReason = "empty content");
+    var error(message7, error(message8, _, code = code8, reason = reason8), errNum = code7, errReason = reason7) = myErr;
+
+    MyError error(message9, error(message10, _, code = code10, reason = reason10), errNum = code9, errReason = reason9) = myErr;
```

Comment:
These tests may fail in the future. See https://github.com/ballerina-platform/ballerina-spec/issues/984. But I'm OK with keeping them for now.